src/nvim/statusline.c

```diff
@@ -451,14 +427,25 @@ void win_redr_custom(win_T *wp, bool draw_winbar, bool draw_ruler)
     // Use 'tabline'.  Always at the first line of the screen.
     stl = p_tal;
     row = 0;
+    opt = "tabline";
     fillchar = ' ';
     attr = HL_ATTR(HLF_TPF);
     maxwidth = Columns;
-    use_sandbox = was_set_insecurely(wp, "tabline", 0);
+  } else if (row != -1) {
+    // Use 'numbercolumn' on the passed row.
+    numbercolumn = true;
+    stl = wp->w_p_nuc;
+    col = wp->w_wincol + wp->w_scwidth + compute_foldcolumn(wp, 0);
+    row += wp->w_winrow + wp->w_winbar_height;
+    opt = "numbercolumn";
+    fillchar = ' ';
+    attr = HL_ATTR(HLF_N);
```

Comment:
Should `'cursorline(opt)' be parsed here and `attr` set to `HLF_CLN` instead if appropriate? I think it could result in unexpected behavior if we don't, even though users can set the highlight with `%#HlGroup#` in the format string as well.

Reply:
I see we can use `drawline:get_line_number_attr()` for that. Could we make it non-static?

Reply:
Hmm it requires a `startrow` and `filler_lines` arg. Would have to pass `startrow` to statusline.c:win_redr_custom()` from `drawscreen.c` and call `diff.c:diff_check()` there. Reasonable?

Reply:
Actually I think it might make more sense to move the `win_redr_custom()` call to somewhere in `win_line()` instead. I think I originally put it in `drawscreen.c` for the vim patch because `win_line()` and `fold_line()` are split there. In that case it can stay static and we can probably re-use the result of the `get_line_number_attr()` call that is already there. Looking into it.

Reply:
> and we can probably re-use the result of the `get_line_number_attr()`

Passing this to `win_redr_custom()` doesn't seem to work. The result is white text, IDK why...

Reply:
This is working now.